File: 0001-two-sum/0001-two-sum.go
/*
Time - O(n)
Space - O(n)
*/
func twoSum(nums []int, target int) []int {
    m:=make(map[int]int)
    ans:=make([]int, 0)
    for i, num:=range nums {
        if val, ok:=m[target-num]; ok {
            ans=append(ans, val)
            ans=append(ans, i)
            return ans
        }
        m[num]=i
    }
    return nil
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash=collections.defaultdict(int)
        for i, num in enumerate(nums):
            if target-num in hash:
                return [i, hash[target-num]]
            hash[num]=i
        return []
        